Whoop, never mind. I have it. It's happening on our PPA builds as well. The relevant sections of buildlog:

Partial output from [PPA buildlog](https://launchpadlibrarian.net/566498966/buildlog_ubuntu-hirsute-amd64.libopenshot_0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1_BUILDING.txt.gz):

<details>
<pre><code>
[ 84%] Building CXX object src/CMakeFiles/openshot.dir/ImageReader.cpp.o
cd "/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu/src" && /usr/bin/c++ -DHAVE_ISFINITE=1 -DLINUX -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DUSE_BABL=1 -DUSE_HW_ACCEL=1 -DUSE_IMAGEMAGICK=1 -Dopenshot_EXPORTS -I"/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu/src/openshot_autogen/include" -I"/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src" -I"/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu/src" -I/usr/include/ImageMagick-6 -I/usr/include/x86_64-linux-gnu/ImageMagick-6 -isystem /usr/include/libopenshot-audio -isystem /usr/include/jsoncpp -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -isystem /usr/include/babl-0.1 -g -O2 -ffile-prefix-map=/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -pedantic -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fPIC -fopenmp -std=c++14 -o CMakeFiles/openshot.dir/ImageReader.cpp.o -c "/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp"
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In constructor ‘zmq::message_t::message_t()’:
/usr/include/zmq.hpp:329:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  329 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In constructor ‘zmq::message_t::message_t(zmq::message_t&&)’:
/usr/include/zmq.hpp:383:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  383 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In destructor ‘zmq::message_t::~message_t()’:
/usr/include/zmq.hpp:396:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  396 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘void zmq::message_t::rebuild()’:
/usr/include/zmq.hpp:405:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  405 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In constructor ‘zmq::context_t::context_t(int, int)’:
/usr/include/zmq.hpp:674:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  674 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘int zmq::context_t::setctxopt(int, int)’:
/usr/include/zmq.hpp:696:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  696 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘void zmq::context_t::close()’:
/usr/include/zmq.hpp:734:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  734 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘void zmq::context_t::shutdown()’:
/usr/include/zmq.hpp:746:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  746 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In constructor ‘constexpr zmq::mutable_buffer::mutable_buffer(void*, size_t)’:
/usr/include/zmq.hpp:945:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  945 |         assert(p != nullptr || n == 0);
      |         ^~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In constructor ‘constexpr zmq::const_buffer::const_buffer(const void*, size_t)’:
/usr/include/zmq.hpp:982:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
  982 |         assert(p != nullptr || n == 0);
      |         ^~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘zmq::recv_result_t zmq::detail::socket_base::recv(zmq::message_t&, zmq::recv_flags)’:
/usr/include/zmq.hpp:1440:13: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
 1440 |             assert(msg.size() == static_cast<size_t>(nbytes));
      |             ^~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘void zmq::socket_t::close()’:
/usr/include/zmq.hpp:1640:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
 1640 |         ZMQ_ASSERT(rc == 0);
      |         ^~~~~~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/10/cassert:44,
                 from /usr/include/zmq.hpp:80,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ZmqLogger.h:25,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/Frame.h:32,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:18:
/usr/include/zmq.hpp: In member function ‘bool zmq::monitor_t::check_event(int)’:
/usr/include/zmq.hpp:1769:9: error: ‘__assert_fail’ was not declared in this scope; did you mean ‘MagickCore::__assert_fail’?
 1769 |         assert(_monitor_socket);
      |         ^~~~~~
In file included from /usr/include/ImageMagick-6/magick/memory_.h:22,
                 from /usr/include/ImageMagick-6/magick/MagickCore.h:125,
                 from /usr/include/ImageMagick-6/Magick++/Include.h:45,
                 from /usr/include/ImageMagick-6/Magick++.h:10,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/MagickUtilities.h:20,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.h:29,
                 from /<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/src/ImageReader.cpp:16:
/usr/include/assert.h:69:13: note: ‘MagickCore::__assert_fail’ declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
make[3]: *** [src/CMakeFiles/openshot.dir/build.make:1021: src/CMakeFiles/openshot.dir/ImageReader.cpp.o] Error 1
make[3]: Leaving directory '/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:1044: src/CMakeFiles/openshot.dir/all] Error 2
make[2]: Leaving directory '/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu'
make[1]: *** [Makefile:185: all] Error 2
make[1]: Leaving directory '/<<BUILDDIR>>/libopenshot-0.2.6+dfsg2+1082+202110311421+daily~ubuntu21.04.1/obj-x86_64-linux-gnu'
dh_auto_build: error: cd obj-x86_64-linux-gnu && make -j1 VERBOSE=1 returned exit code 2
make: *** [debian/rules:15: build] Error 25
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
</code></pre>
</details>

It appears the issue is twofold:

1. ImageMagick's `Magick++/Include.h` header namespaces its inclusion of the C headers from `MagickCore` and `MagickWand`:
    ```c++
    //
    // Include ImageMagick headers into namespace "MagickCore". If
    // MAGICKCORE_IMPLEMENTATION is defined, include ImageMagick development
    // headers.  This scheme minimizes the possibility of conflict with
    // user code.
    //
    namespace MagickCore
    {
    #include <magick/MagickCore.h>
    #include <wand/MagickWand.h>
    #undef inline // Remove possible definition from config.h
    
    #undef class
    }
    ```
   Those  headers then `#include <assert.h>`.
1. ZeroMQ later uses `#include <cassert>`, which depends on `assert.h`. While `assert.h` isn't completely restricted by include guards, a good deal of its functionality is gated on symbols not having already been `#define`d. Since it's a C header and is using the preprocessor, it has no way of knowing that the symbols it already exported are restricted to the `MagickCore` namespace, and it fails to re-define them in the way that ZeroMQ needs.

I'm not sure why this isn't happening to me on Fedora — it could be that I rarely build in Debug mode, or that I'm building with IM7 while Ubuntu/Debian are still on IM6. Maybe they fixed this somewhere down the line.

An alternative workaround would be to always ensure that `<zmq.hpp>` is included before `<Magick++.h>`, but that risks pulling the ZeroMQ headers into classes where we don't  need them just to protect its asserts against Magick++. All in all @troyrollo 's workaround seems simpler, so that's what we'll go with.

I'm going to shuffle around a few `#include`s so that, for instance, the ImageMagick includes in `Frame.h` are replaced with a forward declaration of `Magick::Image` (as that's all it needs), that way the full weight of ImageMagick isn't  being pulled into every class that includes "Frame.h" (which is nearly all of them).

While that might even solve this without the workaround, for extra safety and future-proofing I think it's a good idea to do both. including `<cassert>` in `MagickUtilities.h` costs us almost nothing, and ensures this won't come back to bite us in the future.

--- a/src/ChunkReader.cpp
+++ b/src/ChunkReader.cpp
@@ -10,6 +10,8 @@
 //
 // SPDX-License-Identifier: LGPL-3.0-or-later
 
+#include <fstream>
+
 #include "ChunkReader.h"
 #include "Exceptions.h"
 #include "FFmpegReader.h"

--- a/src/Clip.cpp
+++ b/src/Clip.cpp
@@ -14,14 +14,17 @@
 #include "Exceptions.h"
 #include "FFmpegReader.h"
 #include "FrameMapper.h"
-#ifdef USE_IMAGEMAGICK
-	#include "ImageReader.h"
-	#include "TextReader.h"
-#endif
 #include "QtImageReader.h"
 #include "ChunkReader.h"
 #include "DummyReader.h"
 #include "Timeline.h"
+#include "ZmqLogger.h"
+
+#ifdef USE_IMAGEMAGICK
+    #include "MagickUtilities.h"
+    #include "ImageReader.h"
+    #include "TextReader.h"
+#endif
 
 #include <Qt>
 

--- a/src/FFmpegReader.cpp
+++ b/src/FFmpegReader.cpp
@@ -18,6 +18,7 @@
 #include "FFmpegReader.h"
 #include "Exceptions.h"
 #include "Timeline.h"
+#include "ZmqLogger.h"
 
 #include <thread>    // for std::this_thread::sleep_for
 #include <chrono>    // for std::chrono::milliseconds

--- a/src/Frame.cpp
+++ b/src/Frame.cpp
@@ -10,6 +10,11 @@
 //
 // SPDX-License-Identifier: LGPL-3.0-or-later
 
+#include <thread>    // for std::this_thread::sleep_for
+#include <chrono>    // for std::chrono::milliseconds
+#include <sstream>
+#include <iomanip>
+
 #include "Frame.h"
 #include <OpenShotAudio.h>
 
@@ -27,8 +32,9 @@
 #include <QPointF>
 #include <QWidget>
 
-#include <thread>    // for std::this_thread::sleep_for
-#include <chrono>    // for std::chrono::milliseconds
+#ifdef USE_IMAGEMAGICK
+    #include "MagickUtilities.h"
+#endif
 
 using namespace std;
 using namespace openshot;

--- a/src/Frame.h
+++ b/src/Frame.h
@@ -21,23 +21,27 @@
 	#undef int64
 #endif
 
-
-#include <iomanip>
-#include <sstream>
 #include <queue>
-#include <QApplication>
-#include <QImage>
 #include <memory>
-#include <unistd.h>
-#include "ZmqLogger.h"
+
 #include "ChannelLayouts.h"
 #include "AudioBufferSource.h"
 #include "AudioResampler.h"
 #include "Fraction.h"
+
 #include <OpenShotAudio.h>
+
+#include <QColor>
+#include <QImage>
+
 #ifdef USE_IMAGEMAGICK
-	#include "MagickUtilities.h"
+// Forward declare Magick::Image
+namespace Magick {
+    class Image;
+}
 #endif
+
+class QApplication;
 
 namespace openshot
 {

--- a/src/FrameMapper.cpp
+++ b/src/FrameMapper.cpp
@@ -17,6 +17,7 @@
 #include "FrameMapper.h"
 #include "Exceptions.h"
 #include "Clip.h"
+#include "ZmqLogger.h"
 
 using namespace std;
 using namespace openshot;

--- a/src/Qt/PlayerPrivate.cpp
+++ b/src/Qt/PlayerPrivate.cpp
@@ -13,6 +13,7 @@
 
 #include "PlayerPrivate.h"
 #include "Exceptions.h"
+#include "ZmqLogger.h"
 
 #include <thread>    // for std::this_thread::sleep_for
 #include <chrono>    // for std::chrono milliseconds, high_resolution_clock

--- a/src/Qt/VideoCacheThread.cpp
+++ b/src/Qt/VideoCacheThread.cpp
@@ -12,8 +12,9 @@
 
 #include "VideoCacheThread.h"
 #include "Exceptions.h"
-#include <algorithm>
+#include "ZmqLogger.h"
 
+#include <algorithm>
 #include <thread>    // for std::this_thread::sleep_for
 #include <chrono>    // for std::chrono::milliseconds
 

--- a/src/Qt/VideoPlaybackThread.cpp
+++ b/src/Qt/VideoPlaybackThread.cpp
@@ -12,6 +12,7 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 
 #include "VideoPlaybackThread.h"
+#include "ZmqLogger.h"
 
 namespace openshot
 {

--- a/src/TrackedObjectBBox.cpp
+++ b/src/TrackedObjectBBox.cpp
@@ -11,6 +11,8 @@
 //
 // SPDX-License-Identifier: LGPL-3.0-or-later
 
+#include <fstream>
+
 #include "TrackedObjectBBox.h"
 
 #include "Clip.h"

